File: files/home/.claude/hooks/deny-rm-rf.py
#!/usr/bin/env -S jq -c -f

def RM_RF_PATTERN: "(?i)\\brm\\s+-[a-z]*r[a-z]*f[a-z]*\\s+|\\brm\\s+-[a-z]*f[a-z]*r[a-z]*\\s+";

def command: (.tool_input.command // "");

if .tool_name == "Bash" and (command | test(RM_RF_PATTERN))
then {updatedInput: {command: (command | gsub(RM_RF_PATTERN; "trash "))}}
else empty end

File: files/home/.config/opencode/plugin/deny-rm-rf.js
const RM_RF_PATTERN = /\brm\s+-[a-z]*r[a-z]*f[a-z]*\s+|\brm\s+-[a-z]*f[a-z]*r[a-z]*\s+/i

export const DenyRmRf = async () => {
  return {
    "tool.execute.before": async (input, output) => {
      if (input.tool !== "bash") {
        return
      }

      const command = output?.args?.command ?? ""

      if (RM_RF_PATTERN.test(command)) {
        throw new Error("Refusing to run `rm -rf`. Use `trash` instead.")
      }
    }
  }
}
